File: app/src/main/java/com/example/viredapp/model/Feed.java
package com.example.viredapp.model; ;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Feed {

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("username")
    @Expose
    private String username;
    @SerializedName("m_url")
    @Expose
    private String mUrl;
    @SerializedName("timeStamp")
    @Expose
    private String timeStamp;
    @SerializedName("location")
    @Expose
    private String location;
    @SerializedName("likes_count")
    @Expose
    private Integer likesCount;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getMUrl() {
        return mUrl;
    }

    public void setMUrl(String mUrl) {
        this.mUrl = mUrl;
    }

    public String getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(String timeStamp) {
        this.timeStamp = timeStamp;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public Integer getLikesCount() {
        return likesCount;
    }

    public void setLikesCount(Integer likesCount) {
        this.likesCount = likesCount;
    }

}

File: app/src/main/java/com/example/viredapp/db/feed.kt
package com.example.viredapp.db

import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters
import android.support.annotation.NonNull
import com.example.viredapp.utilities.DateConverter
import com.google.gson.annotations.SerializedName
import java.math.BigInteger
import java.net.URL
import java.util.*

@Entity(tableName = "feed")
data class feed(
        @PrimaryKey(autoGenerate = true)val auto_id:Int,
        @field:SerializedName(value = "id") val id:Int,
        @field:SerializedName(value = "username")@NonNull val username:String,
        @field:SerializedName(value = "m_url")@NonNull val mUrl:String,
        @field:SerializedName(value = "timeStamp")@NonNull val timeStamp:String,
        @field:SerializedName(value = "location")val location:String?,
        @field:SerializedName(value = "likes_count")@NonNull val likes_count:String
)

